src/main/scala/com/ing/wbaa/airlock/proxy/provider/kafka/EventProducer.scala

```diff
@@ -0,0 +1,43 @@
+package com.ing.wbaa.airlock.proxy.provider.kafka
+
+import akka.Done
+import akka.stream.ActorMaterializer
+import com.ing.wbaa.airlock.proxy.config.KafkaSettings
+import org.apache.kafka.clients.producer.{ KafkaProducer, ProducerConfig, ProducerRecord }
+import org.apache.kafka.common.serialization.StringSerializer
+
+import scala.concurrent.duration.SECONDS
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+
+trait EventProducer {
+  import scala.collection.JavaConverters._
+
+  protected[this] implicit val kafkaSettings: KafkaSettings
+
+  protected[this] implicit val materializer: ActorMaterializer
+
+  protected[this] implicit val executionContext: ExecutionContext
+
+  private lazy val config: Map[String, Object] =
+    Map[String, Object](
+      "bootstrap.servers" -> kafkaSettings.bootstrapServers,
+      ProducerConfig.RETRIES_CONFIG -> kafkaSettings.retries,
+      ProducerConfig.RECONNECT_BACKOFF_MS_CONFIG -> kafkaSettings.retriesBackOff,
+      ProducerConfig.RECONNECT_BACKOFF_MAX_MS_CONFIG -> kafkaSettings.retriesBackOffMax
+    )
+
+  def kafkaProducer: KafkaProducer[String, String] = new KafkaProducer(config.asJava, new StringSerializer, new StringSerializer)
+
+  def sendSingleMessage(event: String, topic: String): Future[Done] = {
+    val sentF = kafkaProducer.send(new ProducerRecord[String, String](topic, event))
+    Try(
+      sentF.get(3, SECONDS)
+    ) match {
+        case Success(r) =>
+          kafkaProducer.close()
```

Comment:
I don't think we really want to close the producer.

https://stackoverflow.com/questions/39917046/how-to-reconnect-kafka-producer-once-closed

> The KafkaProducer.send method is asynchronous, it returns a Future[RecordMetadata]. If you close immediatly after sending, you have a race condition, and due to the buffering nature of the KafkaProducer your message might never get sent.
> 
> If your producer is being used throughout the lifetime of your application, don't close it, and let it die once the application terminates. As said in the documentation, the producer is safe to used in a multi-threaded environment and hence you should re-use the same instance.